src/client/server.spec.ts

```diff
@@ -1,18 +1,91 @@
-import type { FastifyInstance } from 'fastify';
+import { jest } from '@jest/globals';
+import type { FastifyInstance, LightMyRequestResponse } from 'fastify';
 import { CloudEvent } from 'cloudevents';
+import type { DeliveryOptions } from '@relaycorp/relaynet-pohttp/build/main/lib/client.js';
+import { getModelForClass } from '@typegoose/typegoose';
+import type { Connection } from 'mongoose';
+import { generatePDACertificationPath } from '@relaycorp/relaynet-testing';
+import {
+  CertificationPath,
+  type Channel,
+  Parcel,
+  PrivateEndpointConnParams,
+  ServiceMessage,
+  SessionKeyPair,
+} from '@relaycorp/relaynet-core';
+import { addDays, addSeconds, differenceInSeconds, formatISO, subDays, subSeconds } from 'date-fns';
 
-import { setUpTestPohttpClient } from '../testUtils/pohttpClient.js';
 import { HTTP_STATUS_CODES } from '../utilities/http.js';
 import { CE_ID, CE_SOURCE } from '../testUtils/eventing/stubs.js';
 import { postEvent } from '../testUtils/eventing/cloudEvents.js';
+import { mockSpy } from '../testUtils/jest.js';
+import { type MockLogSet, partialPinoLog } from '../testUtils/logging.js';
+import { PeerEndpoint } from '../models/PeerEndpoint.model.js';
+import { KEY_PAIR_SET, PEER_ADDRESS, PEER_KEY_PAIR } from '../testUtils/awala/stubs.js';
 
-import { PohttpClientProblemType } from './PohttpClientProblemType.js';
+const mockDeliverParcel = mockSpy(
+  jest.fn<
+    (
+      recipientInternetAddressOrURL: string,
+      parcelSerialized: ArrayBuffer | Buffer,
+      options?: Partial<DeliveryOptions> | undefined,
+    ) => Promise<void>
+  >(),
+);
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+class PoHTTPInvalidParcelError extends Error {}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+class PoHTTPClientBindingError extends Error {}
+
+jest.unstable_mockModule('@relaycorp/relaynet-pohttp', () => ({
+  deliverParcel: mockDeliverParcel,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  PoHTTPInvalidParcelError,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  PoHTTPClientBindingError,
+}));
+
+const { setUpTestPohttpClient } = await import('../testUtils/pohttpClient.js');
 
 describe('makePohttpClient', () => {
+  const expiry = addDays(Date.now(), 5);
+  const cloudEventData = {
+    id: CE_ID,
+    source: CE_SOURCE,
+    type: 'testType',
+    subject: 'peerId',
+    datacontenttype: 'test/content-type',
+    expiry: formatISO(expiry),
+    data: 'test data',
+  };
+
   const getTestServerFixture = setUpTestPohttpClient();
   let server: FastifyInstance;
-  beforeEach(() => {
-    ({ server } = getTestServerFixture());
+  let logs: MockLogSet;
+  let dbConnection: Connection;
+  let sessionPrivateKey: CryptoKey;
```

Comment:
```suggestion
  let peerSessionPrivateKey: CryptoKey;
```

It's important to qualify it. We have two.